src/SNPN/Common/INotificationService.cs

```diff
@@ -1,9 +1,10 @@
 ﻿using System.Xml.Linq;
+using SNPN.Model;
 
 namespace SNPN.Common
 {
 	public interface INotificationService
 	{
-		void QueueNotificationData(NotificationType type, string notificationUri, XDocument content = null, NotificationGroups group = NotificationGroups.None, string tag = null, int ttl = 0);
+		void QueueNotificationData(NotificationType type, string notificationUri, Post post, NotificationMatchType matchType, string title, string message, NotificationGroups group = NotificationGroups.None, int ttl = 0);
```

Comment:
The content generation was passed into a later function so that the Post, Notification Match Type (reply/mention/keyword) can be passed to queued notification items for SB functionality.